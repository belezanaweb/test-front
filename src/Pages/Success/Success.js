import React from 'react'
import Products from '../../Components/Products/index'
import Total from '../../Components/Total/index'
import { Container, Title } from '../../globalstyles'
import { Paymentbox, Image } from './styles'
import imagesuccess from '../../Assets/success.png'

export default function Success() {
  return (
    <Container>
      <Image src={imagesuccess}></Image>
      <Title>PAGAMENTO</Title>
      <Paymentbox>
        <span>****.****.****.1234</span>
        <span>JOSÃ‰ DA SILVA</span>
        <span>05/2019</span>
      </Paymentbox>
      <Title>Produtos</Title>
      <Products />
      <Total />
    </Container>
  )
}
