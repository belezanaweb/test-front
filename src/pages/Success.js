import React from 'react';

const Success = () => (
  <div>
    success page
  </div>
);

export default Success;
