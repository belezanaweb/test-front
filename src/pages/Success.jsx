export function Success() {
  return <h1>Success</h1>
}