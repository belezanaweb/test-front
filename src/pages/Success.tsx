import React from 'react'

import ProductList from '@/components/products/ProductList'
import ResumeValues from '@/components/resume/ResumeValues'
import { useCart } from '@/contexts/cart'

function Success (): React.ReactElement | null {
  const { cart } = useCart()

  if (!cart) {
    return null
  }

  return (
    <>
      <ProductList />
      <ResumeValues />
    </>
  )
}

export default Success
