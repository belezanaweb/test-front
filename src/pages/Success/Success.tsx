import React from 'react'

export const Success = () => <h1>Success</h1>
