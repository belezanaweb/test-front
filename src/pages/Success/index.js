import React from 'react'

const Success = () => {
  return (
    <div>
      <h1>Success</h1>
    </div>
  )
}

export default Success
