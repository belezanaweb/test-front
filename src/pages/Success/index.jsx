import { Header } from '../../components/Header'
import { Content } from '../../components/Content'
import { ProductCard } from '../../components/ProductCard'
import { PriceList } from '../../components/PriceList'
import { headerItems } from '../../types'

import { useLocation } from 'react-router-dom'

const Success = () => {
  const { state } = useLocation()

  return (
    <>
      <Header activeItem={headerItems.CONFIRMAÇÃO} headerItems={Object.values(headerItems)} />
      <Content title="Pagamento" type="white">
        Lorem Ipsum
      </Content>
      <Content title="Produtos" type="white">
        {state.data.items.map((item) => (
          <ProductCard
            hideDetails
            key={item.product.sku}
            product={{
              name: item.product.name,
              image: item.product.imageObjects[0].small,
              price: item.product.priceSpecification.price
            }}
          />
        ))}
      </Content>
      <Content>
        <PriceList
          prices={{
            subTotal: state.data.subTotal,
            shippingTotal: state.data.shippingTotal,
            discount: state.data.discount,
            total: state.data.total
          }}
        />
      </Content>
    </>
  )
}

export default Success
