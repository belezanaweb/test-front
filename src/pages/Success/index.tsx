import React from 'react'
import InfoWrapper from '../../components/InfoWrapper'
import Menu from '../../components/Menu'
import * as S from './styles'
import { useCartDetails } from '../../hooks/useCartDetails'
import Product from '../../components/Product'
import Subtotal from '../../components/Subtotal'
import Loader from '../../components/Loader'
import { formatCreditCardNumber, formatExpirationDate } from '../../util/format'
import CartSuccess from '../../components/CartSuccess'
import Empty from '../../components/Empty'
import MediaMatch from '../../components/MediaMatch'
import Dropdown from '../../components/Dropdown'

export function Success() {
  const {
    discount,
    shippingTotal,
    subTotal,
    total,
    items,
    mounted,
    userPaymentInfo
  } = useCartDetails()
  return (
    <>
      {!mounted ? (
        <Loader />
      ) : (
        <>
          {Object.keys(userPaymentInfo).length === 0 ? (
            <>
              <S.Wrapper>
                <S.MenuWrapper>
                  <Menu activeLink="/success" />
                </S.MenuWrapper>
                <S.EmptyWrapper>
                  <Empty />
                </S.EmptyWrapper>
              </S.Wrapper>
            </>
          ) : (
            <>
              <S.Wrapper>
                <S.MenuWrapper>
                  <Menu activeLink="/success" />
                </S.MenuWrapper>
                <S.CartWrapper>
                  <S.StatusPaymentWrapper>
                    <S.StatusWrapper>
                      <CartSuccess />
                    </S.StatusWrapper>
                    <InfoWrapper title="PAGAMENTO">
                      <S.UserPaymentInfoWrapper>
                        <S.Text>
                          {userPaymentInfo.creditCard &&
                            formatCreditCardNumber(userPaymentInfo.creditCard)}
                        </S.Text>
                        <S.Text>
                          {userPaymentInfo.nameInCard.toUpperCase()}
                        </S.Text>
                        <S.Text>
                          {userPaymentInfo.expirationDate &&
                            formatExpirationDate(
                              userPaymentInfo.expirationDate
                            )}
                        </S.Text>
                      </S.UserPaymentInfoWrapper>
                    </InfoWrapper>
                    <MediaMatch greaterThan="medium">
                      <S.PaymentInfo>
                        <Subtotal
                          shippingTotal={shippingTotal}
                          subTotal={subTotal}
                          total={total}
                          discount={discount}
                        />
                      </S.PaymentInfo>
                    </MediaMatch>
                  </S.StatusPaymentWrapper>
                  <S.ProductPaymentInfoWrapper>
                    <MediaMatch greaterThan="medium">
                      <S.AcessibilityWrapper>
                        <Dropdown />
                      </S.AcessibilityWrapper>
                    </MediaMatch>
                    <InfoWrapper title="PRODUTOS">
                      <S.Product>
                        {items?.map((item) => (
                          <Product
                            key={item.product.sku}
                            name={item.product.name}
                            imageUrl={item.product.imageObjects[0].small}
                          />
                        ))}
                      </S.Product>
                    </InfoWrapper>
                    <MediaMatch lessThan="medium">
                      <S.PaymentInfo>
                        <Subtotal
                          shippingTotal={shippingTotal}
                          subTotal={subTotal}
                          total={total}
                          discount={discount}
                        />
                      </S.PaymentInfo>
                    </MediaMatch>
                  </S.ProductPaymentInfoWrapper>
                </S.CartWrapper>
              </S.Wrapper>
              <MediaMatch lessThan="medium">
                <S.AcessibilityWrapper>
                  <Dropdown />
                </S.AcessibilityWrapper>
              </MediaMatch>
            </>
          )}
        </>
      )}
    </>
  )
}
