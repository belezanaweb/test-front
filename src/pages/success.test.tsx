import React from 'react';
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import axios from 'axios';

import mock from '../store/models/__mocks__/cart.json';

import store from '../store';

import App from '../App';
import Success from './success';

jest.mock('axios');

beforeEach(() => {
  jest.resetAllMocks();
});

describe('Success Page', () => {
  it('should render success page', async () => {
    axios.get = jest.fn().mockResolvedValue({ data: mock });

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/success"]}>
          <App>
            <Success />
          </App>
        </MemoryRouter>
      </Provider>
    );

    await Success;

    expect(container).toMatchSnapshot();
  });
});
