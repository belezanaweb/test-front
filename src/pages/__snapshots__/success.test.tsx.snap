// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success Page should render success page 1`] = `
<div>
  <div
    class="sc-fzoLag jRHNZI"
  >
    <nav
      class="sc-fzqBZW kdufKf"
    >
      <ol>
        <li
          class="sc-fzqNJr gREgXq"
          title="Sacola"
        >
          <a>
            Sacola
          </a>
        </li>
        <li
          class="sc-fzqNJr gREgXq"
          title="Pagamento"
        >
          <a>
            Pagamento
          </a>
        </li>
        <li
          class="sc-fzqNJr beoRzK"
          title="Confirmação"
        >
          <a>
            Confirmação
          </a>
        </li>
      </ol>
    </nav>
    <div
      class="sc-fzoyAV biEWFk"
    >
      <div
        class="sc-fznyAO jLApxi"
      >
        <div
          class="sc-fznKkj laAbkH"
        />
        <span
          class="sc-AxgMl sc-fznZeY iTQCPi"
        >
          Compra Efetuada com Sucesso
        </span>
      </div>
      <span
        class="sc-AxgMl sc-fzokOt kTHzsh"
      >
        Pagamento
      </span>
      <div
        class="sc-AxiKw sc-AxhUy daxRmo"
      >
        <div
          class="sc-Axmtr hBVTto"
        >
          <div
            class="sc-AxjAm gTlBjH"
            direction="row"
          >
            <div
              class="sc-AxiKw gmkaqg"
            >
              <span
                aria-label="card number"
                class="sc-AxgMl sc-AxheI cyIPIf"
              >
                ****.****.****.undefined
              </span>
            </div>
            <div
              class="sc-AxiKw gmkaqg"
            >
              <span
                aria-label="name"
                class="sc-AxgMl sc-AxheI cyIPIf"
              />
            </div>
            <div
              class="sc-AxiKw gmkaqg"
            >
              <span
                aria-label="expiring date"
                class="sc-AxgMl sc-AxheI cyIPIf"
              />
            </div>
          </div>
        </div>
      </div>
      <span
        class="sc-AxgMl sc-fzokOt kTHzsh"
      >
        Produtos
      </span>
      <div
        class="sc-AxiKw sc-AxhUy hkJMki"
      >
        <div
          class="sc-fzqNqU jbHngq"
        />
      </div>
    </div>
  </div>
</div>
`;
